stdlib/public/Platform/musl.modulemap

```diff
@@ -2,110 +2,876 @@
+
+module stdc_stdalign [system] {
+  header "stdalign.h"
+  export *
+}
+
+module stdc_stdarg [system] {
+  header "stdarg.h"
+  export *
+}
+
+module stdc_stdatomic [system] {
+  header "stdatomic.h"
+  export *
+}
+
+module stdc_stdbool [system] {
+  header "stdbool.h"
+  export *
+}
+
+module stdc_stddef [system] {
+  header "stddef.h"
+  export *
+}
+
+module stdc_stdint [system] {
+  header "stdint.h"
+  export *
+}
+
+module stdc_time [system] {
+  config_macros _POSIX_SOURCE, _POSIX_C_SOURCE, _XOPEN_SOURCE, _GNU_SOURCE, _BSD_SOURCE
+  header "time.h"
+  export *
+}
+
+module stdc_tgmath [system] {
+  header "tgmath.h"
+  export *
+}
+
+module stdc_complex [system] {
   header "complex.h"
+  export *
+}
+
+module stdc_ctype [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "ctype.h"
+  export *
+}
+
+module stdc_errno [system] {
+  config_macros _GNU_SOURCE
   header "errno.h"
+  export *
+}
+
+module stdc_fenv [system] {
   header "fenv.h"
-  header "float.h"
+  export *
+}
+
+module stdc_inttypes [system] {
   header "inttypes.h"
-  header "iso646.h"
-  header "limits.h"
+  export *
+}
+
+module stdc_locale [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "locale.h"
+  export *
+}
+
+module stdc_math [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _XOPEN_SOURCE
   header "math.h"
-  header "pty.h"
+  export *
+}
+
+module stdc_setjmp [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "setjmp.h"
+  export *
+}
+
+module stdc_signal [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "signal.h"
-  header "stdarg.h"
-  header "stdbool.h"
-  header "stddef.h"
-  header "stdint.h"
+  export *
+}
+
+module stdc_stdio [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _LARGEFILE64_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "stdio.h"
+  export *
+}
+
+module stdc_stdlib [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _LARGEFILE64_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "stdlib.h"
+  export *
+}
+
+module stdc_stdnoreturn [system] {
+  header "stdnoreturn.h"
+  export *
+}
+
+module stdc_string [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
   header "string.h"
-  header "tgmath.h"
-  header "time.h"
-  header "utmp.h"
+  export *
+}
+
+module stdc_threads [system] {
+  header "threads.h"
+  export *
+}
+
+module stdc_uchar [system] {
+  header "uchar.h"
+  export *
+}
 
-  // POSIX
+module stdc_wchar [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
+  header "wchar.h"
+  export *
+}
+
+module stdc_wctype [system] {
+  config_macros _BSD_SOURCE, _GNU_SOURCE, _POSIX_C_SOURCE, _POSIX_SOURCE, _XOPEN_SOURCE
+  header "wctype.h"
+  export *
+}
+
+// POSIX
```

Comment:
I don't know how we want to organize the rest of these. One module per header can start to cause performance issues, but having too many headers in a module can also cause performance issues. Is there some kind of logical grouping we might be able to use like filesystem, networking, etc?

Reply:
I _guess_ these are less likely than the C library headers to suffer from the circularity issues created by the builtin modules and the C++ library, so we probably could group them. I just broke them all out for now.

Reply:
I started looking at the Android module map that's being worked on to see how things were grouped there. I'm wondering if we should try to make things match. Not sure that's a goal right now, but it's an interesting question.